tensorflow_model_optimization/python/core/sparsity_tf2/sparse_utils.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Convenience functions for sparse training."""
+
+class Bernouilli(tf.keras.initializers.Initializer):
+  """
+  Initialization distributio following a Bernouilli process..
+  """
+
+  def __init__(self, p):
+    """
+    p: probability parameter of success (i.e. 1).
+    """
+    if not (p >= 0. and p <= 1.):
+      raise ValueError('p parameter must be a valid probability, i.e. in [0, 1].')
+    self.p = p
+
+  def get_config(self):
+    return {'p': self.p}
+
+  @classmethod
+  def from_config(cls, config):
+    return cls(**config)
+
+  def __call__(self, shape, dytpe=tf.dtypes.float32):
+    """Number of zeros = np.ceil(probability * size) in expectation."""
+    probs = tf.zeros(shape=list(shape)) + self.p
+    uniform = tf.random.uniform(shape)
+    initial = tf.less(uniform, probs)
+
+    return tf.cast(initial, dtype=dtype)
+
+class PermuteOnes(tf.keras.initializers.Initializer):
+  """
+  Initialization of a deterministically sparse matrix.
+  This initializer takes in an input ratio and sets exactly
+  that ratio of the mask entries as ones  leaving the rest as zeros.
+  The ones are detministically, randomly permmuted across the tensor.
```

Comment:
I would get rid of `deterministic` because I'm not sure this is actually guaranteeing determinism if you don't specify a seed in `__call__`. (And passing this in directly as an initializer object won't use seed)